### ISSUE TITLE
BUG: filter with a multi-grouping including a datetimelike fails

### ISSUE THREAD

**Issue by jreback** [pydata/pandas#10114]
this was changed in #9921 (worked prior to 0.16.0)
cc @evanpw

so using a multiple grouper where one item is a datetimelike fails as the ``name in self.indices`` fails. Instead ``.get_group()`` can be used to fix

```
In [1]: df = DataFrame({'A' : np.arange(5), 'B' : ['foo','bar','foo','bar','bar'], 'C' : Timestamp('20130101') })

In [2]: df
Out[2]: 
   A    B          C
0  0  foo 2013-01-01
1  1  bar 2013-01-01
2  2  foo 2013-01-01
3  3  bar 2013-01-01
4  4  bar 2013-01-01

In [3]: df.groupby(['B','C']).filter(lambda x: True)
Out[3]: 
Empty DataFrame
Columns: [A, B, C]
Index: []

In [4]: df.groupby(['B']).filter(lambda x: True)
Out[4]: 
   A    B          C
0  0  foo 2013-01-01
1  1  bar 2013-01-01
2  2  foo 2013-01-01
3  3  bar 2013-01-01
4  4  bar 2013-01-01

In [5]: df.groupby(['C']).filter(lambda x: True)
Out[5]: 
   A    B          C
0  0  foo 2013-01-01
1  1  bar 2013-01-01
2  2  foo 2013-01-01
3  3  bar 2013-01-01
4  4  bar 2013-01-01
```

**Comment by edparcell** [comment]
In the meantime, the following worked for me:
```python
simple = pd.DataFrame({
        'A': pd.Series([pd.to_datetime('2015-01-01')]), 
        'B': pd.Series([1], dtype=np.float64),
        'C': pd.Series([3], dtype=np.float64)})
simple.set_index(['A', 'B'])[simple.groupby(['A', 'B']).apply(lambda x: True)].reset_index()
```
Not as performant, but it works. Drops index, so would have to do reset_index before and set_index after to retain index.

**Comment by jreback** [comment]
closed by #10124
